homeassistant/components/unmanic/config_flow.py

```diff
@@ -0,0 +1,121 @@
+"""Adds config flow for Unmanic integration."""
+from unmanic_api import Unmanic
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_TIMEOUT,
+    CONF_VERIFY_SSL,
+    __version__,
+)
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import (
+    DEFAULT_PORT,
+    DEFAULT_SSL,
+    DEFAULT_TIMEOUT,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+)
+
+
+class UnmanicConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Unmanic."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        self._errors = {}
+
+        if user_input is not None:
+            configuration_already_exists = await self._config_exists(
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+                self.hass,
+            )
```

Comment:
Why manually check if the configuration already exists? You can set the unique id and use the helpers. My only concern would be that a host is not a super unique address, is there nothing in the config / setup that is more unique?

```suggestion
            await self.async_set_unique_id(user_input[CONF_HOST] + user_input[CONF_PORT])

            self._abort_if_unique_id_configured()
```

Reply:
With this change, when it's aborted due to a duplication configuration, it looks like this:
![Aborted](https://i.imgur.com/GkclOM0.png)

Is there any way to catch this as an exception or something similar and display a more normal-looking error, similar to how it was before (or how the `cannot_connect` error looks)?
![Cannot connect error](https://i.imgur.com/XcAPc2a.png)